src/catalog/CatalogPage.test.tsx

```diff
@@ -206,36 +206,6 @@ describe('CatalogPage', () => {
     });
   });
 
-  it('should call fetchData with search query when search button is clicked', async () => {
-    const mockFetchData = jest.fn();
-    mockUseCourseListSearch.mockReturnValue({
-      isLoading: false,
-      isError: false,
-      data: mockCourseListSearchResponse,
-      fetchData: mockFetchData,
-      isFetching: false,
-    });
-
-    render(<CatalogPage />);
-
-    const searchField = screen.getByPlaceholderText(messages.searchPlaceholder.defaultMessage);
-    await userEvent.type(searchField, 'python');
-
-    const searchButton = screen.getByRole('button', { name: 'search submit search' });
-    await userEvent.click(searchButton);
-
-    await waitFor(() => {
-      expect(mockFetchData).toHaveBeenCalledWith(
-        expect.objectContaining({
-          pageIndex: DEFAULT_PAGE_INDEX,
-          pageSize: DEFAULT_PAGE_SIZE,
-          filters: [],
-          searchString: 'python',
-        }),
-      );
-    });
-  });
-
```

Comment:
With the internal search button we don't have a search button when text has been entered, as it is replaced with a clear button. We already have "should call fetchData with search query when search is submitted," so when moving to an internal search button this test feels redundant.